### PR TITLE
Add FeetechMotorsBus, SO-100, Moss-v1

### PR DESCRIPTION
**Pull request by Cadene**
## What this does

**Feetech motor bus, SO-100, Moss-v1**

- [Feature] Add `lerobot/common/robot_devices/motors/feetech.py`
- [Feature] Add `lerobot/configs/robot/so100.yaml` and `lerobot/configs/robot/moss.yaml`
- [Feature] Add `examples/10_use_so100.md` and `examples/11_use_moss.md`
- [Refactor] Move find motors bus port to `lerobot/scripts/find_motors_bus_port.py`
- [Refactor] Move configure motors ID+baudrate to `scripts/configure_motor.py`

**Auto-calibration**

- [Refactor] Move calibration to `lerobot/common/robot_devices/robots/dynamixel_calibration.py`
- [Feature] Add `lerobot/common/robot_devices/robots/feetech_calibration.py` with autocalibration for SO-100 and Moss-v1 follower arms

**Minor**

- [Feature] Add `read_with_motor_ids(...num_retry=NUM_READ_RETRY` and `write_with_motor_ids(...num_retry=NUM_WRITE_RETRY`) to set them to a low value of retries during `configure_motor.py` and thus speedup configuration process

See tutorial: https://github.com/huggingface/lerobot/blob/1d92acf1e3d6123874605d8048e110d58a383868/examples/10_use_so100.md

## How it was tested

- Ran full calibration
- Recored some episodes

https://huggingface.co/spaces/lerobot/visualize_dataset?dataset=cadene%2Fmoss_debug&episode=0
https://huggingface.co/spaces/lerobot/visualize_dataset?dataset=cadene%2Fso100_debug&episode=0

### CODE REVIEW

File: lerobot/common/robot_devices/robots/manipulator.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import torch
 
 from lerobot.common.robot_devices.cameras.utils import Camera
-from lerobot.common.robot_devices.motors.dynamixel import (
+from lerobot.common.robot_devices.motors.feetech import (
```
Comment (Cadene) [inline lerobot/common/robot_devices/robots/manipulator.py:13]: TODO: remove before merging

File: tests/test_cameras.py [hunk 2]
```diff
@@ -131,7 +134,8 @@ def test_camera(request, robot_type):
     del camera
```
Comment (jess-moss) [inline tests/test_cameras.py:136]: I removed: `@require_robot`-- should I put this back?

**Review by ramkumarkoppu — APPROVED** [review]

**Review by Cadene — COMMENTED** [review]
review with Simon

**Review by Cadene — COMMENTED** [review]
Reviewed with Simon

**Pull request merged**